components/org.wso2telco.analytics.sparkUdf/src/main/java/org/wso2telco/analytics/sparkUdf/service/BillUpdaterService.java

```diff
@@ -1,89 +1,93 @@
+import org.killbill.billing.client.model.Account;
 import org.killbill.billing.client.model.Invoice;
 import org.killbill.billing.client.model.InvoiceItem;
 import org.wso2telco.analytics.sparkUdf.configProviders.ConfigurationDataProvider;
 
-import java.math.BigDecimal;
-import java.util.Date;
-import java.util.List;
-import java.util.UUID;
-
 /**
  * @author dilan
  */
 public class BillUpdaterService {
 
 
-    public void updateBillForMonth(String accountId, Date date, String description, int amount) {
-        KillBillHttpClient killBillHttpClient = null;
-        KillBillClient killBillClient = null;
-
-        try {
-            ConfigurationDataProvider dataProvider = ConfigurationDataProvider.getInstance();
-
-            killBillHttpClient = new KillBillHttpClient(dataProvider.getUrl(), dataProvider.getUname(),
-                    dataProvider.getPassword(), dataProvider.getApiKey(), dataProvider.getApiSecret());
-
-            killBillClient = new KillBillClient(killBillHttpClient);
-
-            List<Invoice> invoices = killBillClient.getInvoicesForAccount(UUID.fromString(accountId), RequestOptions.empty());
-            Invoice invoice = getInvoiceForMonthFromList(invoices, date);
-            if (invoice != null) {
-                updateInvoice(invoice, description, amount, killBillClient);
-            }
-
-        } catch (Exception e) {
-
-        } finally {
-            if (killBillClient != null) {
-                killBillClient.close();
-            }
-            if (killBillHttpClient != null) {
-                killBillHttpClient.close();
-            }
-        }
-    }
-    /*
-     *
+	private static ConfigurationDataProvider dataProvider=null;
+	private static KillBillHttpClient killBillHttpClient;
+	private static KillBillClient killBillClient;
+	private static UUID invoiceItemId=null;
+
+	@SuppressWarnings("deprecation")
+	public String l(String accountId,int year,int month,String description,int amount){
+		Account account=null;
+		try {
+			dataProvider=ConfigurationDataProvider.getInstance();
+
+			killBillHttpClient= new KillBillHttpClient(dataProvider.getUrl(),
+					dataProvider.getUname(),
+					dataProvider.getPassword(),
+					dataProvider.getApiKey(),
+					dataProvider.getApiSecret());
+
+			killBillClient = new KillBillClient(killBillHttpClient);
+
+			List<Invoice> invoices=killBillClient.getInvoicesForAccount(UUID.fromString(accountId));
+			Invoice invoice=getInvoiceForMonthFromList(invoices, year,month);
+			
+			if (invoice!=null) {
+				invoiceItemId=updateInvoice(invoice, description, amount);	
+			}
+
+		}catch (Exception e) {
+			return "Bill was not updated";
+		}finally{
+			if (killBillClient!=null) {
+				killBillClient.close();
+			}
+			if (killBillHttpClient!=null) {
+				killBillHttpClient.close();
+			}
+		}
+		return invoiceItemId.toString();
+		
+	}
+
+	private Invoice getInvoiceForMonthFromList(List<Invoice> invoices,int year,int month) {
+		// TODO Auto-generated method stub
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.wso2telco.com:9000/coding_rules#rule_key=squid%3AS1135)